tests/src/python/test_qgsserver_wms_getcapabilities_group_name.py

```diff
@@ -0,0 +1,87 @@
+"""QGIS Unit tests for QgsServer GetCapabilities group name attribute exclusion.
+
+From build dir, run: ctest -R PyQgsServerWMSGetCapabilities -V
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+"""
+__author__ = 'Tomas Straupis'
+__date__ = '2023-10-13'
+__copyright__ = 'Copyright 2023, The QGIS Project'
+
+import os
+import urllib.parse
+
+from qgis.testing import unittest
+from test_qgsserver import QgsServerTestBase
+
+
+class TestQgsServerWMSGetCapabilities(QgsServerTestBase):
+    """QGIS Server WMS Tests for GetFeatureInfo request"""
+
+    # Set to True to re-generate reference files for this class
+    regenerate_reference = False
+
+    def setUp(self):
+        super().setUp()
+        # First project has "skipping name tag for groups" setting set to false (default)
+        self.project_with_name = os.path.join(self.testdata_path, "wms_group_test1.qgs")
+        # Second project has "skipping name tag for groups" setting set to true
+        self.project_without_name = os.path.join(self.testdata_path, "wms_group_test2.qgs")
+
+    def test_wms_getcapabilities_with(self):
+        r = make_capabilities_request(self, self.project_with_name)
+        f = str(r).find('<Name>layer_group</Name>')
```

Comment:
For testing inclusion or exclusion of a string (or bytes) in the response you can use this construct directly:

```suggestion
        self.assertIn(b'<Name>layer_group</Name>', r, 'attribute <name> should be specified for a layer group')
```

no need to use find.  Conversion to str from bytes can also be avoided.

Same for the other tests.

Reply:
Oh, I see. Thank you! This looks much better!